path-manager/app/controllers/PathManagerController.scala

```diff
@@ -28,37 +28,55 @@ object PathManagerController extends Controller {
   }
 
   def registerExistingPath(id: Long) = Action { request =>
-    request.body.asJson.map(_.as[PathRecord]).map { submission =>
-      if (id != submission.identifier) {
+    val maybeSubmission = request.body.asJson.map(_.as[PathRecord])
+    maybeSubmission match {
+      case (Some(path)) if path.identifier != id => {
         PathOperationErrors.increment
         Logger.warn("registerExistingPath failed, identifier in url and body do not match")
         BadRequest("identifier in url and body do not match")
-      } else if (submission.`type` != "canonical") {
-        PathOperationErrors.increment
-        Logger.warn("registerExistingPath failed, only canonical paths can be updated at present")
-        BadRequest("only canonical paths can be updated at present")
-      } else {
-        PathStore.register(submission) match {
+      }
+
+      case Some(path) if path.`type` == "alias" => {
+        PathStore.registerAlias(path) match {
+          case Left(error) => BadRequest(error)
+          case Right(records) => {
+            PathMigrationRegistrations.increment
+            argoOk(Json.toJson(records))
+          }
+        }
+      }
+      case Some(path) if path.`type` == "canonical"  => {
+        PathStore.registerCanonical(path) match {
           case Left(error) => BadRequest(error)
           case Right(records) => {
             PathMigrationRegistrations.increment
             argoOk(Json.toJson(records))
           }
         }
       }
-    } getOrElse{
-      PathOperationErrors.increment
-      Logger.warn("registerExistingPath failed, unable to parse PathRecord from request body")
-      BadRequest("unable to parse PathRecord from request body")
+      case Some(_)  => {
+        PathOperationErrors.increment
+        Logger.warn("registerExistingPath failed, only canonical and alias paths can be updated at present")
+        BadRequest("only canonical paths can be updated at present")
+      }
+      case None => {
+        PathOperationErrors.increment
+        Logger.warn("registerExistingPath failed, unable to parse PathRecord from request body")
+        BadRequest("unable to parse PathRecord from request body")
+      }
+
     }
-  }
 
-  def updateCanonicalPath(id: Long) = Action { request =>
+  }
 
+  def updateCanonicalPath(id: Long /*, formAlias: Boolean*/) = Action { request =>
     val submission = request.body.asFormUrlEncoded.get
     val newPath = submission("path").head
 
-    PathStore.updateCanonical(newPath, id) match {
+    val update = if (request.getQueryString("formAlias") contains "true") {PathStore.updateCanonicalWithAlias _} else {PathStore.updateCanonical _}
+    //Can't get this to bind, try after Tom's big upgrade
```

Comment:
when you say it isn't binding? is that from postman, or from Composer UI (if so, query params need to be essentially 'allow-listed' in composer-backend as per https://github.com/guardian/flexible-content/pull/3555/commits/b860c5b72a06d34b9937a67080002ad8a01e12ff#diff-f0283f6e89cc87b6244d311f848fb391R526 see https://github.com/guardian/flexible-content/pull/3555/commits/66a4cc9d33f9f8610c621957481f8cce986cc594)

Reply:
I've been doing the HTTP by hand on this, and play wasn't binding the route like I expected. I'm going to revisit on your later play build. (Obviously before merging)